### PR TITLE
Test datepicker

### PR DESCRIPTION
**Pull request by kurtwall**
Interacting with the datepicker turned out to be extremely timing sensitive. The test specifically needs to wait for elements to become visible on the page, at least when testing against a local instance.

### CODE REVIEW

**Comment by kurtwall** [comment]
```
[firefox #0-0] Spec: /home/kwall/Work/TolaActivity/test/js/tests/collecteddata_datepicker.js
[firefox #0-0] Running: firefox
[firefox #0-0]
[firefox #0-0] Collected data datepicker
[firefox #0-0]
[firefox #0-0]     using clicked dates
[firefox #0-0]       ✓ does not display a date automatically when selected
[firefox #0-0]       ✓ closes the datepicker window after selecting a date
[firefox #0-0]
[firefox #0-0]
[firefox #0-0] 2 passing (32s)
[firefox #0-0]

[chrome #0-0] Spec: /home/kwall/Work/TolaActivity/test/js/tests/collecteddata_datepicker.js
[chrome #0-0] Running: chrome
[chrome #0-0]
[chrome #0-0] Collected data datepicker
[chrome #0-0]
[chrome #0-0]     using clicked dates
[chrome #0-0]       ✓ does not display a date automatically when selected
[chrome #0-0]       ✓ closes the datepicker window after selecting a date
[chrome #0-0]
[chrome #0-0]
[chrome #0-0] 2 passing (29s)
[chrome #0-0]
```

**Comment by kurtwall** [comment]
Obviously, the biggest change is the additional tests and the page object. I also got sloppy and included a mish-mash of tweaks and cleanups unrelated to those changes.

**Comment by kurtwall** [comment]
3128824 makes the login logic into a utility function and DRYs out the test code nicely.

File: test/js/tests/collecteddata_datepicker.js [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    it.skip('assigns given value to correct period if program lifetime includes date (quarterly)')
+    it.skip('assigns given value to correct period if program lifetime includes date (monthly)')
+    it.skip('displays warning if given date is outside program lifetime (all target periods)')
+
+    it('does not do the hokey-pokey', function() {
+        expect(true)
+    })
+  })
+
+})
```
Comment (sanjuroj) [inline test/js/tests/collecteddata_datepicker.js]: It seems like the act of choosing a program and then choosing an indicator and then opening up the collected data modal window is common to most of the tests.  Could this series of actions be abstracted out to its own utility function to DRY things out a bit?

**Pull request merged**